src/EditorFeatures/Core/Shared/Utilities/ForegroundThreadAffinitizedObject.cs

```diff
@@ -32,7 +32,7 @@ internal static ForegroundThreadData CreateDefault(ForegroundThreadDataKind defa
         {
             var kind = ForegroundThreadDataInfo.CreateDefault(defaultKind);
 
-            return new ForegroundThreadData(Thread.CurrentThread, new SynchronizationContextTaskScheduler(SynchronizationContext.Current), kind);
+            return new ForegroundThreadData(Thread.CurrentThread, SynchronizationContext.Current == null ? TaskScheduler.Default : new SynchronizationContextTaskScheduler(SynchronizationContext.Current), kind);
```

Comment:
Actually, should this be TaskScheduler.Current? I don't presume we're using this much in the /build case, but Default _does_ mean we're always picking the thread pool. Current would mean it might be right (and might still be wrong, but whatever.)

Reply:
https://msdn.microsoft.com/en-us/library/system.threading.tasks.taskscheduler.current(v=vs.110).aspx

Remarks
When not called from within a task, Current will return the Default scheduler.

..

so in this case, it will be Default anyway?

Reply:
If we run this inside of an xunit unit test, then it might pick up the xunit scheduler or something which _might_ be what we want there.

Reply:
Although I guess in that case SynchronizationContext.Current won't be null anyways. 🤷‍♂️